gemini/src/main/java/com/techempower/cache/EntityStore.java

```diff
@@ -1327,18 +1326,33 @@ else if (field.isAnnotationPresent(Right.class))
                 throw new RuntimeException(
                     "Cannot create CachedRelation from @Relation definition class without specifying @Left and @Right Identifiables.");
               }
+
+              final Left leftAnnotation = left.getAnnotation(Left.class);
+              final Right rightAnnotation = right.getAnnotation(Right.class);
               
               // We're ready to register this CachedRelation.  The table name will
               // be inferred from the class name.  The left and right column names
               // will use the name of the parameters.
               // Don't register it more than once.
-              if (getRelation((Class<EntityRelationDescriptor<Identifiable, Identifiable>>)clazz) == null)
-              {
-                register(CachedRelation.of((Class<? extends Identifiable>)left.getType(), (Class<? extends Identifiable>)right.getType()).
-                    table(clazz.getSimpleName()).
-                    leftColumn(left.getName()).
-                    rightColumn(right.getName()),
-                    clazz);
+              if (getRelation((Class<EntityRelationDescriptor<Identifiable, Identifiable>>)clazz) == null) {
+                final CachedRelation.Builder<?, ?> builder = CachedRelation.of(
+                        (Class<? extends Identifiable>) left.getType(),
+                        (Class<? extends Identifiable>) right.getType());
+                if (StringHelper.isNonEmpty(annotation.table()))
```

Comment:
The elements of an annotation can't be null, so you don't need `StringHelper` here.

```
if (!annotation.table.isEmpty())
```

Reply:
I honestly did not know this.